distributedtracing/distributedtracing.go

```diff
@@ -0,0 +1,60 @@
+package distributedtracing
+
+import (
+	"context"
+
+	"github.com/open-policy-agent/opa/logging"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	"go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.7.0"
+)
+
+// Enabled defines whether OpenTelemtry distributed tracing is enabled
+var Enabled bool
```

Comment:
Hmm package-level vars for configuration feel a bit like a shortcut 🤔 Is the alternative unbearable? I can't tell without trying myself.

Reply:
I couldn't find an easy way to access it from https://github.com/open-policy-agent/opa/blob/main/topdown/http.go.
Any suggestions are more than welcome :slightly_smiling_face:

Reply:
We'd thread it through builtincontext, I think.

Reply:
generally what i have done if i want to conditionally enable tracing is default to the noop tracer https://github.com/open-telemetry/opentelemetry-go/blob/main/trace/noop.go#L27 if tracing configuration is not present then there is no need to add logic on whether to instrument tracing or not, everything just gets instrumented.

Reply:
Anything the builtin `http.send` needs from the evaluator would be passed down via the [`BuiltinContext`](https://github.com/open-policy-agent/opa/blob/v0.34.2/topdown/builtins.go#L34-L53). So we could add fields there.

The question I can't answer right now is what we'd want to have there: 
- [`otelhttp.Option`](https://pkg.go.dev/go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp#Option) I think if we include that, it'll be enough to have all the tracer config stuff passed along?
- Do we care for anything evaluation-related? Would this be a descendent of some span that corresponds to the incoming request...? If so, how is that information passed around?

Sorry if this is vague, I really need you input here. Just trying to wrap my mind around how it would work and be most useful.

Reply:
Pass []otelhttp.Option along in stead of using global trace provider